features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/EditParameter.jsx

```diff
@@ -181,22 +216,50 @@ export default function EditParameter(props) {
                                     },
                                 }}
                             >
-                                {['number', 'integer'].map((dataType) => (
+                                {getSupportedDataTypes(version, parameter.in).map((dataType) => (
                                     <MenuItem value={dataType} dense>
                                         {capitalizeFirstLetter(dataType)}
                                     </MenuItem>
                                 ))}
                             </Select>
-                            <FormHelperText>Select schema type</FormHelperText>
+                            <FormHelperText>
+                                <FormattedMessage
+                                    id={'Apis.Details.Resources.components.operationComponents.EditParameter.'
+                                    + 'select.schema.data.type'}
+                                    defaultMessage='Select the Schema Type'
+                                />
+                            </FormHelperText>
                         </FormControl>
                     </Grid>
                     <Grid item md={6}>
-                        <FormControl fullWidth margin='dense' variant='outlined' className={classes.formControl}>
-                            <InputLabel htmlFor='edit-parameter-format'>Format</InputLabel>
+                        <FormControl
+                            fullWidth
+                            margin='dense'
+                            variant='outlined'
+                            className={classes.formControl}
+                            disabled={parameter.schema
+                                ? iff(
+                                    parameter.schema.type === 'boolean' || parameter.schema.type === 'object',
+                                    true,
+                                    false,
+                                )
+                                : iff(
```

Comment:
When do we get parameter properties outside a **schema**

Reply:
in swagger 2.0 the type can be in the root level of the parameter object.